lib/api/src/grpc/conversions.rs

```diff
@@ -1607,6 +1607,76 @@ impl From<segment::data_types::order_by::StartFrom> for StartFrom {
     }
 }
 
+impl TryFrom<MaxOptimizationThreads> for rest::MaxOptimizationThreads {
```

Comment:
That's actually a nice idea. Since it is a common pattern in all the conversions, it might be a good idea to generalize it into a conversion error which takes the type it failed to convert from, and has the same error message for all of them.

It would be best to do it in a separate PR, though :)